### PR TITLE
Make some LUKS tests nondestructive, and run basic ones in tmt

### PR DESCRIPTION
**Pull request by martinpitt**

### CODE REVIEW

**Comment by martinpitt** [comment]
I have no idea how the changes here could affect the miniscule rendering of the checkbox, but it [happened](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19262-20230830-145031-88cb4227-fedora-38-storage/log.html) [twice](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19262-20230830-195756-88cb4227-fedora-38-storage/log.html) now. I'll check tomorrow morning if that happens on main already.

File: test/verify/check-storage-luks [hunk 1]
```diff
@@ -197,6 +193,96 @@ class TestStorageLuks(storagelib.StorageCase):
...
+                     "old_passphrase": "vainu-reku-toma-rolle-kaja",
+                     "mount_point": "/run/foo"})
+        self.content_row_wait_in_col(1, 2, "ext4 filesystem (encrypted)")
+
+
+class TestStorageLuksDestructive(storagelib.StorageCase):
+
+    # LUKS uses memory hard PBKDF, 1 GiB is not enough; see https://bugzilla.redhat.com/show_bug.cgi?id=1881829
+    provision = {
+        "0": {"memory_mb": 1536}
```
Comment (mvollmer) [inline test/verify/check-storage-luks:283]: Is this still needed with the new default?

**Review by mvollmer — DISMISSED** [review]
Thanks! I am still not a fan of non-destruciveness in general, but I can't argue with the speedup they give us.

File: test/verify/check-storage-luks (hunk 1 above)
Comment (martinpitt) [reply test/verify/check-storage-luks:283]: I only changed the size of the "global" machines, i. e. for nondestructive tests. This is destructive, so it's still needed.

**Comment by martinpitt** [comment]
@mvollmer TBH I consider the speedup a nice side effect. My main motivation is that they are easier to debug and iterate on (especially with watch + rsync mode), and allow running them in tmt, so that we can start gating cryptsetup/LUKS changes.

File: test/verify/check-storage-luks (hunk 1 above)
Comment (mvollmer) [reply test/verify/check-storage-luks:283]: Ah, thanks.

**Comment by mvollmer** [comment]
> My main motivation is that they are easier to debug and iterate on (especially with watch + rsync mode),

Mostly because they are faster, right?  I iterate on destructive (and non-destructive) tests with watch mode and putting a suitable `m.upload` into the test itself. Same convenience, just slower.

> and allow running them in tmt, so that we can start gating cryptsetup/LUKS changes.

Indeed, but wouldn't it be nice if tmt would be able to run destructive tests? That's one of the things that keeps me from taking it serious.

**Comment by mvollmer** [comment]
> I have no idea how the changes here could affect the miniscule rendering of the checkbox, but it [happened](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19262-20230830-145031-88cb4227-fedora-38-storage/log.html) [twice](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19262-20230830-195756-88cb4227-fedora-38-storage/log.html) now. I'll check tomorrow morning if that happens on main already.

I had to ignore some checkboxes in another PR as well: https://github.com/cockpit-project/cockpit/pull/18867, hmm.  These are not SVGs, but natively rendered by the Browser.

**Comment by martinpitt** [comment]
Happened a [third time](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19262-20230831-064604-88cb4227-fedora-38-storage/log.html#54), so I gave up and just pushed the new references. :man_shrugging:

**Review by mvollmer — APPROVED** [review]

**Pull request merged**